msl/equipment/resources/vaisala/ptu300.py

```diff
@@ -0,0 +1,199 @@
+        """Vaisala Barometer PTU300 series.
+        Device manual: https://docs.vaisala.com/v/u/M210855EN-D/en-US
+
+        Do not instantiate this class directly. Use the :meth:`~.EquipmentRecord.connect`
+        method to connect to the equipment.
+
+        :param record: A record from an :ref:`equipment-database`.
+        """
+        super(PTU300, self).__init__(record)
+
+        self.set_exception_class(VaisalaError)
+        self.rstrip = True
+
+        self.ID = record.serial
+
+        self.stop_run_mode()
+
+        self._units = {}
+        self.pressure_modules = set()
+        self.info = self.device_info(show=False)
+        self.check_serial()
+
+    def check_serial(self) -> str:
+        """Get the device ID (serial) number and check it agrees with the equipment record.
+
+        :return: the serial number as a string
+        """
+        # can use reply = self.query("*9900SN") but the serial number is also in the device info
+        sn = self.info["Serial number"]
+        if not sn == self.ID:
+            self.raise_exception(f"Inconsistent serial number: expected {self.ID} but received {sn}")
+        return sn
+
+    def device_info(self, show: bool = True) -> dict:
+        """Prints information about the Vaisala device.
+
+        :param show: bool for whether to display the device info as log messages
+        :return: a dictionary of the device information
+        """
+        info = {}
+        break_keys = {
+            'PTU30': "module 2",
+            'PTB33': "module 4",
+        }
+        break_key = "TBC"
+        self.write("?")
+        while True:
+            ok = self.read()
+            if show:
+                self.log_info(ok)
+            try:
+                key, val = ok.split(': ')
+                info[key.strip()] = val.strip()
+                if 'baro' in val.lower():
+                    self.pressure_modules.add(val)
+                if break_key in key.lower():
+                    break
+            except ValueError:
+                model, version = ok.split(" / ")
+                info['Model'] = model
+                break_key = break_keys[model[:-1]]
+                info['Version'] = version
+
+        return info
+
+    def set_units(self, desired_units: dict[str, str]) -> None:
+        """Set units of specified quantities. Note that only one pressure unit is used at a time for the PTU300 series.
+
+        :param desired_units: dictionary of quantities and units as specified in the instrument manual on page 22.
+            These may include the following (available options depend on the barometer components):
+            Pressure quantities: P, P3h, P1, P2, QNH, QFE, HCP, ...
+            Pressure units: hPa, psi, inHg, torr, bar, mbar, mmHg, kPa, Pa, mmH2O, inH2O
+            Temperature quantity: T
+            Temperature units: °C, °F
+            Humidity quantity: RH
+            Humidity unit: %RH
+        """
```

Comment:
This docstring won't render the way you hope it does in the docs. See other comment in this review that included a screenshot.